workspace/workspace.go

```diff
@@ -38,6 +39,42 @@ func New(dir string) (Workspace, error) {
 	return Workspace{Dir: dir}, nil
 }
 
+// Exercises is the user's exercises within the workspace.
+// It doesn't include exercises by other users downloaded
+// from the site.
+func (ws Workspace) Exercises() ([]Exercise, error) {
+	exercises := []Exercise{}
+
+	topInfos, err := ioutil.ReadDir(ws.Dir)
+	if err != nil {
+		return nil, err
+	}
+	for _, topInfo := range topInfos {
+		if !topInfo.IsDir() {
+			continue
+		}
+
+		if topInfo.Name() == "users" {
+			continue
+		}
+
+		subInfos, err := ioutil.ReadDir(filepath.Join(ws.Dir, topInfo.Name()))
+		if err != nil {
+			return nil, err
+		}
+
+		for _, subInfo := range subInfos {
+			if !subInfo.IsDir() {
+				continue
+			}
+
+			exercises = append(exercises, Exercise{Track: topInfo.Name(), Slug: subInfo.Name(), Root: ws.Dir})
```

Comment:
This is a great point.

I actually think we need both. For the `exercism doctor` command I want to check everything that is _placed in the correct location_, but which is missing a solution file.

How about if we add a method `HasExerciseMetadata()`, and then have two finder methods, one that finds all the _potential_ exercises, and one which actually checks that the metadata exists, and only returns those?

Reply:
> I actually think we need both. For the exercism doctor command I want to check everything that is placed in the correct location, but which is missing a solution file.

Ah good point. 

> How about if we add a method HasExerciseMetadata(), and then have two finder methods, one that finds all the potential exercises, and one which actually checks that the metadata exists, and only returns those?

Thinking about this a little, two methods makes sense to start. One method that finds all directories and the second one that calls the first method but applies a filter to remove any directory that does not contain a metadata file. 

If we ever need to change the find all method then we could break out the generic exercise finder into a third unexported function and have the two exported methods call that under the hood. I'm tempted to say go with three to start but I can go either way. 

Do we need to filter out dotifles?

Reply:
> Do we need to filter out dotifles?

I think it would make sense to do so preemptively.

> Thinking about this a little, two methods makes sense to start. One method that finds all directories and the second one that calls the first method but applies a filter to remove any directory that does not contain a metadata file.

I like it.

Thanks for the help thinking this through!